packages/webview_flutter/webview_flutter_platform_interface/lib/src/types/web_resource_request.dart

```diff
@@ -0,0 +1,43 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+/// Defines the parameters of the web resource request from the associated request.
+///
+/// Platform specific implementations can add additional fields by extending
+/// this class.
+///
+/// This example demonstrates how to extend the [WebResourceRequest] to
+/// provide additional platform specific parameters.
+///
+/// When extending [WebResourceRequest] additional parameters should always
+/// accept `null` or have a default value to prevent breaking changes.
+///
+/// ```dart
+/// class AndroidWebResourceRequest extends WebResourceRequest {
+///   WebResourceRequest._({
+///     required WebResourceRequest request,
+///   }) : super(
+///     uri: request.uri,
+///   );
+///
+///   factory AndroidWebResourceRequest.fromWebResourceRequest(
+///     WebResourceRequest request, {
+///     Map<String, String> headers,
+///   }) {
+///     return AndroidWebResourceRequest._(request, headers: headers);
+///   }
+///
+///   final Map<String, String> headers;
+/// }
+/// ```
+@immutable
+class WebResourceRequest {
```

Comment:
We should have trivial tests of the new classes that test that the constructors populate the params as expected. It's not inherently that useful, but it creates a place where tests can be easily and obviously added if code is ever added here in the future (e.g., if someone adds equality logic).